src/bank.rs

```diff
@@ -300,7 +306,10 @@ impl Bank {
                 Err(_) => RpcSignatureStatus::GenericFailure,
             };
             if status != RpcSignatureStatus::SignatureNotFound {
-                self.check_signature_subscriptions(&tx.signatures[0], status);
```

Comment:
The match on lines 302-307 can move out of the bank into into rpc_* too, with that ideally we can remove the bank dependency on `RpcSignatureStatus` entirely

Reply:
👍